### PR TITLE
Rewrite FrameTimeVisualizer in performance page

### PR DESCRIPTION
**Pull request by kenzieschmoll**
- The frame time visualizer now has bars that are sized proportionately to the time a frame phase consumed (build, layout, paint, etc. time). 
Screen Shot 2022-08-01 at 3 42 55 PM

- The concept of selection was removed since it will not be used
- Added tooltips and polished the UI for narrow widths
adaptive-frame-time-viz

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_analysis_model.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_analysis_model.dart:85]: do not review this part here - this should be reviewed in https://github.com/flutter/devtools/pull/4327, and then will merge here once that PR lands.

File: packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart
Comment (CoderDake) [inline packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart:194]: Remove extra f here

File: packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart
Comment (CoderDake) [inline packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart:227]:
```
final adjustedBlockWidths = unadjustedBlockWidths.map(b => math.max(b,  _FramePhaseBlock.minBlockWidth));
adjustedBlockWidths[widestBlockIndex] -= adjustment; 
```
I think you can simplify this with a map and an index modification :)

File: packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart
Comment (CoderDake) [inline packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart:211]: if all blocks are less than the min width, then our widest block will end up being less than the min width. Is that ok based on the context with these block resizes?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart:211]: Below where we calculate the adjustedWidth, we adjust so that the width is at least `_FramePhaseBlock.minBlockWidth`:

`adjustedWidth = math.max(adjustedWidth, _FramePhaseBlock.minBlockWidth);`

So it won't end up being less than the min width after the adjustment. And if we overflow when all blocks are ` _FramePhaseBlock.minBlockWidth` wide, then so be it, because that means the screen is more narrow than we intend to support.

File: packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart:227]: subtracting the adjustment needs to happen before we perform the `max` calculation so that we do not end up with an `adjustedWidth < _FramePhaseBlock.minBlockWidth` for the widest block index. 

I can simplify this though with another call to `max` when assigning `adjustedBlockWidths[widestBlockIndex]`.

File: packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart]: done

**Review by CoderDake — APPROVED** [review]

**Pull request merged**
